virtualbox: 6.0.8 -> 6.0.10

NeQuissimus opened the pull request:
###### Motivation for this change
Update VirtualBox

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @flokli @svanderburg

NeQuissimus committed:
virtualbox: 6.0.8 -> 6.0.10

NeQuissimus commented:
@GrahamcOfBorg test virtualbox

flokli commented:
@NeQuissimus we should now be able to re-enable UART in the virtualbox tests, as the upstream bug should be fixed according to the release notes:

> Serial: fixed guru meditation when raw mode is enabled (bug [18632](https://www.virtualbox.org/ticket/18632))

see https://github.com/NixOS/nixpkgs/pull/60943 for context.

cc @ambrop72

flokli commented:
Hm, the tests still seem to fail when enabling uart:

[p2qjm5ncv6n0zw0gv7fk2fjiglj0qmvz-vm-test-run-virtualbox-systemd-detect-virt.drv.log](https://github.com/NixOS/nixpkgs/files/3488808/p2qjm5ncv6n0zw0gv7fk2fjiglj0qmvz-vm-test-run-virtualbox-systemd-detect-virt.drv.log)
[b2vx4dhb8rvgmqy0yns1gl6v9blpl9zh-vm-test-run-virtualbox-simple-gui.drv.log](https://github.com/NixOS/nixpkgs/files/3488809/b2vx4dhb8rvgmqy0yns1gl6v9blpl9zh-vm-test-run-virtualbox-simple-gui.drv.log)
[9j9sfd3cak25zzzhsxx61l08r045k1dx-vm-test-run-virtualbox-net-hostonlyif.drv.log](https://github.com/NixOS/nixpkgs/files/3488810/9j9sfd3cak25zzzhsxx61l08r045k1dx-vm-test-run-virtualbox-net-hostonlyif.drv.log)
[872ix9w0vg7izkm0ni6gsg15jdb8imjm-vm-test-run-virtualbox-simple-cli.drv.log](https://github.com/NixOS/nixpkgs/files/3488811/872ix9w0vg7izkm0ni6gsg15jdb8imjm-vm-test-run-virtualbox-simple-cli.drv.log)

@ambrop72, any idea?

ambrop72 commented:
@flokli VirtualBox is just broken in master currently, see https://github.com/NixOS/nixpkgs/issues/65564. I guess it will work on 19.03.

ambrop72 commented:
@flokli At least on the first log I see the signal 6 error. Anyway did you try without enabling the uart?

flokli commented:
yeah, it's both broken with and without UART. I fear we need to fix #65564 first.

ambrop72 commented:
We could try it out on 19.03 though.

flokli commented:
Yeah, or cherry-pick it on top of `f79fd2e826dd95b3b64839d3e0bec8ae1dfab17e~1`…

But I guess for merging it to 19.03, we need to fix unstable first.

ambrop72 committed:
tests/virtualbox: Clean up unnecessary things.

Always enable the UART because the VirtualBug bug that required running without the UART was fixed in 6.0.10. Stop using an old kernel version because the tests work with the default kernel.

(cherry picked from commit ae93571e8d04cebd69491a789d902d6481e05d3f)

flokli commented:
@NeQuissimus Can this be rebased on top of latest master?

ambrop72 commented:
Could you add this cleanup of tests to this PR?
https://github.com/ambrop72/nixpkgs/commits/virtualbox-cleanup

flokli commented:
@GrahamcOfBorg test virtualbox

flokli commented:
Switched my system to this commit, and also ran the tests successfully.

Thanks!
